### DISCUSSION TITLE
plot several datasets in the same image

### DISCUSSION THREAD

**Discussion by danlipsa** [Alpine-DAV/ascent#1227] [General]
Hi all,
Is it possible to render several datasets in the same image? For instance, the following image from SENSEI oscillators miniapp renders (with ParaView) three datasets:  for the 'mesh' dataset we volume render the 'data' field, for the 'particle' dataset we use a glyph plot for the 'velocity' and for the 'oscillators' dataset we just render the point positions using spheres. How can I produce something similar using ascent? Thanks!
![RenderView2_000040](https://github.com/Alpine-DAV/ascent/assets/6180727/aedd2b2d-fdc8-4d4e-ae77-1c050dc241ca)

**Comment by cyrush** [comment]
Hi @danlipsa, yes you can plot multiple meshes.

Each independent `mesh` should be published as topology (which can have its own distinct coords and fields). 
Since they all co exist in the same tree, they do need to have unique names.

From there you can create an Ascent scene that has multiple plots, which will be combined into a single image. 

Here is a simple example:
https://ascent.readthedocs.io/en/latest/Actions/Examples.html#example-of-rendering-multiple-topologies

One caveat is that you can only do one volume rendering b/c combining multiple volume renders is not well defined. 

Hope this helps and let me know if you have follow up questions.

**Reply by danlipsa** [reply]
Thanks @cyrush It makes sense. So the way fields are defined, you cannot duplicate field names even if they are attached to a different topology?

So, from the field name you can find out everything you need.

**Reply by cyrush** [reply]
That is correct. Since all the fields live next to each other in the tree, they need unique names.

If there is ambiguity  -- a common approach is to simply expand fields with more descriptive names, such as:

```
mesh_density
particle_density
```
